### DISCUSSION TITLE
How to Include Compiled DLLs from IKVM in a NuGet Package?

### DISCUSSION THREAD

**Discussion by Foorcee** [ikvmnet/ikvm#608] [Q&A]
Hello,

I'm currently facing an issue with including DLLs compiled via IKVM in a NuGet package.

I have a .NET project that references some .jar files via IKVM using IkvmReference. When I use dotnet publish, the compiled .jars and .dlls are correctly included in the output directory. However, when I try to publish the project as a NuGet package using dotnet pack, these compiled DLLs are not included in the resulting .nupkg file.

**Comment by wasabii** [comment]
https://github.com/ikvmnet/ikvm?tab=readme-ov-file#notice-to-project-owners

Unless it's a private nuget repository, in which case you can do whatever you want.

**Comment by BiologyTools** [comment]
As wasabii said you shouldn't publish Java libraries to NuGet. However if you use them locally just edit your project file to include the DLL into the lib folder.
```
  <ItemGroup>
    <None Update="bioformats_package.dll">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
         <Pack>True</Pack>
	 <PackagePath>\lib\net8.0</PackagePath>
    </None>
  </ItemGroup>
```
